include/fe/fe_interface.h

```diff
@@ -570,6 +570,19 @@ class FEInterface
                           const unsigned int j,
                           const Point & p);
 
+  /**
+   * Fills \p dphi with the derivatives of the \f$ i^{th} \f$ shape
+   * function at point \p p in direction \p j.
+   */
+  template<typename OutputType>
+  static void shape_derivs(const FEType & fe_t,
+                           const Elem * elem,
+                           const unsigned int i,
+                           const unsigned int j,
+                           const std::vector<Point> & p,
+                           std::vector<OutputType> & dphi,
+                           const bool add_p_level = true);
```

Comment:
Nah, we've had that as an option in many of our FE and FEInterface APIs for a while.  IIRC I tacked it on to one of the pre-existing "take elem->p_level() into account" APIs at one point because no way was I going to add Yet Another Overload; the fact that it actually makes the expected behavior way less confusing was just a nice side bonus.